staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/marshal.go

```diff
@@ -91,6 +146,39 @@ func (s *JSONSchemaPropsOrStringArray) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+func (s JSONSchemaPropsOrStringArray) MarshalCBOR() ([]byte, error) {
+	if len(s.Property) > 0 {
+		return cbor.Marshal(s.Property)
+	}
+	if s.Schema != nil {
+		return cbor.Marshal(s.Schema)
+	}
+	return cbor.Marshal(nil)
+}
+
+func (s *JSONSchemaPropsOrStringArray) UnmarshalCBOR(data []byte) error {
+	switch {
+	case len(data) == 0, data[0] == cborNullValue:
+		return nil
```

Comment:
Hm, the JSON implementation doesn't look like it implements the conventional no-op for null, and instead zeroes out all fields.

I see that you added test cases for UnmarshalJSON that include nulls, but I also noticed that the tests are set up so that the objects under test are always struct fields. I suspect this means that none of the unmarshal tests are actually covering the path where the argument is `[]byte("null")`. Could you check? I think we want the Unmarshal tests, at least, to be calling UnmarshalX directly on the object under test.